webapp/src/main/java/life/qbic/views/login/LoginLayout.java

```diff
@@ -143,53 +104,47 @@ private void styleFormLayout() {
         "shadow-l",
         "min-width-300px",
         "max-width-15vw");
-    contentLayout.add(
-        layoutTitle,
-        errorMessage,
-        confirmationInformationMessage,
-        email,
-        password,
-        loginButton,
-        forgotPasswordButton,
-        registerSpan);
   }
 
-  private void createEmailField() {
-    email = new EmailField("Email");
+  private Span initRegisterSpan() {
+    RouterLink routerLink = new RouterLink("REGISTER", UserRegistrationLayout.class);
+    return new Span(new Text("Need an account? "), routerLink);
   }
 
-  private void createPasswordField() {
-    password = new PasswordField("Password");
+  public void clearErrors() {
+    errorContainer.setVisible(false);
+    errorContainer.removeAll();
   }
 
-  private void createLoginButton() {
-    loginButton = new Button("Log In");
+  public void showError(ErrorMessage errorMessage) {
+    errorContainer.add(new ErrorMessage(errorMessage.title(), errorMessage.message()));
+    errorContainer.setVisible(true);
   }
 
-  private void styleLoginButton() {
-    loginButton.setWidthFull();
-    loginButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
+  public void showInformation(InformationMessage message) {
+    informationContainer.add(new InformationMessage(message.title(), message.message()));
+    informationContainer.setVisible(true);
```

Comment:
What's the reason for recreating the message instead of using the one in the parameter?
NIT: I still think that since the Messages themselves are components to be used in the view their creation belongs in the view. 
So you could also provide the title and message as Strings from the handler

Reply:
they are components of the view but the handler populates it. We could introduce a `Message` record that only delivers the values to the view but I think that adds complexity we do not need right now.